enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -99,259 +116,211 @@ This is often required for industries such as the US public sector, healthcare o
 
 ### Implementation Details
 
-#### Assumptions
-
-* A pipeline shall support at least one output.
-* A pipeline may support multiple outputs.
-* Outputs that require secrets which are not created by the cluster-logging-operator shall be created and managed by the administrator of the endpoint
-* Secrets shall have keys of: `tls.crt`, `tls.key`, `ca-bundler.crt` which point to the respective certificates for which they represent. Secrets shall have the key `shared_key` for use when using `forward` in a secure manner
-* The operator will only support a single instance of log forwarding named `instance` for at least the tech preview release
-* Log forwarding is enabled for tech preview when the instance of `clusterloggging` is annotated with: `clusterlogging.openshift.io/logforwardingtechpreview: enabled`
+* A pipeline associates multiple input names with multiple output names.
+* Users cannot define new inputs (for initial release)
+* Users can define new outputs, output configuration includes:
+  - `type` (e.g. `Syslog`, `Fluent`)
+  - `destination` URL for the endpoint.
+  - `secretRef` referring to a secret object used for secure connections.
+    - secrets must be in the `openshift-logging` namespace (for GA)
+    - a secretRef with no namespace defaults to the `openshift-logging` namespace.
+    - Secrets not created by the cluster-logging-operator shall be created and managed by the administrator of the endpoint
+* Forwarding is enabled when a `ClusterLogForwarder` is created, and disabled when it is deleted.
+  - With no `ClusterLogForwarder`: the default Elasticsearch instance is *enabled* (status quo)
+  - With a `ClusterLogForwarder`: the default Elasticsearch instance is *disabled* unless explicitly enabled by a `pipeline` with output`Default`.
 
 #### Security
-* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: `tls.crt`, `tls.key`, `ca-bundler.crt`
-* The secret must exist in the `openshift-logging` namespace
-* Endpoints defined without a secret will use unsecure communication and must explicitly declare `insecure: true` in the output specifification
 
-#### Log Message Routing
-* Undeclared source types (e.g. `logs.infra`) will be dropped if they are not defined in `spec.pipelines` entry
-* Declared source types are routed to the specified output
+* Server-authenticated TLS is enabled if `destination` is a secure URL (e.g. 'https:')
```

Comment:
shouldn't this require the `ca.crt` entry for the secret?

Reply:
By ''server authentication" I mean browser-style auth - we require the remote destination to have a valid cert but we don't provide one ourselves. For "client auth" you do need the cert as you say. Is there a better term we can use in this context to avoid confusion? Client/server auth are the common HTTP terms, but people who think mostly about "presumed secure"  private networks can be thrown by the idea a server authenticating itself to clients - I was the first time ;)

Reply:
ack, i was thinking of us serving up for some reason